Assets/BossRoom/Scripts/Client/UI/Lobby/CreateLobbyUI.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+using BossRoom.Scripts.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BossRoom.Scripts.Client.UI
+{
+    public class CreateLobbyUI : MonoBehaviour
+    {
+        public const string k_DefaultIP = "127.0.0.1";
+        public const int k_DefaultPort = 9998;
+
+        [SerializeField] private InputField m_LobbyNameInputField;
+        [SerializeField] private CanvasGroup m_IPConnectionCanvasGroup;
+        [SerializeField] private InputField m_IPInputField;
+        [SerializeField] private InputField m_PortInputField;
+        [SerializeField] private GameObject m_LoadingImage;
+        [SerializeField] private Toggle m_IPToggle;
+        [SerializeField] private Toggle m_UnityRelayToggle;
+        [SerializeField] private Toggle m_IsPrivate;
+        [SerializeField] private CanvasGroup m_CanvasGroup;
+
+        private LobbyUIMediator m_LobbyUIMediator;
+
+        private OnlineMode m_OnlineMode;
+
+        private void Awake()
+        {
+            EnableIPHostUI();
+            m_IPToggle.onValueChanged.AddListener(IPRadioRadioButtonPressed);
+            m_UnityRelayToggle.onValueChanged.AddListener(UnityRelayRadioRadioButtonPressed);
+        }
+
+        [Inject]
+        private void InjectDependencies(LobbyUIMediator lobbyUIMediator)
+        {
+            m_LobbyUIMediator = lobbyUIMediator;
+        }
+
+        private void IPRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            EnableIPHostUI();
+        }
+
+        private void UnityRelayRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            EnableUnityRelayUI();
+        }
+
+        private void EnableUnityRelayUI()
+        {
+            m_IPConnectionCanvasGroup.alpha = 0;
+            m_IPConnectionCanvasGroup.blocksRaycasts = false;
+            m_IPConnectionCanvasGroup.interactable = false;
+
+            m_LoadingImage.SetActive(false);
+        }
+
+        private void EnableIPHostUI()
+        {
+            m_IPConnectionCanvasGroup.alpha = 1;
+            m_IPConnectionCanvasGroup.blocksRaycasts = true;
+            m_IPConnectionCanvasGroup.interactable = true;
+
+            m_PortInputField.text = k_DefaultPort.ToString();
```

Comment:
This line is duplicated, and there might be a bug here. In the UI, I don't see the default port, only "PORT" written, and I can only write non-alphanumeric characters inside this textbox (looks like an issue with the Sanitize method).
The IP textbox similarily displays "IP Address", but I can write anything inside the textbox, including letters and non-alphanumeric characters.

Reply:
Fixed